packages/graphback-runtime-knex/src/KnexDBDataProvider.ts

```diff
@@ -28,25 +28,24 @@ export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements G
     this.baseType = baseType;
   }
 
-  public async create(data: Type): Promise<Type> {
+  public async create(data: Type, context: any): Promise<Type> {
```

Comment:
Should context be strongly typed to GraphbackContext?

Reply:
The interfaces:
1. https://github.com/aerogear/graphback/blob/623426121c3573d28730c903d7949e7bbac5f3ea/packages/graphback-core/src/runtime/GraphbackDataProvider.ts
2. https://github.com/aerogear/graphback/blob/623426121c3573d28730c903d7949e7bbac5f3ea/packages/graphback-core/src/runtime/GraphbackCRUDService.ts 

are typed already.

Reply:
Yep in the interfaces, but they are not typed in the class, which is where they are used. If you were to make a typo you would not know because it is `any`.

Reply:
Yes, not sure how that will play with the type defs we have in the interface. 

Just a note, adding a type here will make the tests hard to maintain: This file for example, if we add something e.g `config` to the contex but not using it in `DataProvider`, we'll have to properly add it in all tests files so that the compiler is happy https://github.com/aerogear/graphback/blob/83ca6c904d80b151b97fae1cc309ee172afe886c/packages/graphback-runtime-knex/tests/data/KnexDbDataProviderTest.ts 

Just to get historical reasons, the context is `any` to allow user custom things to be added in context right?

Reply:
Ah, you can do `context: Partial<GraphbackContext>`, making all fields optional.

I think the context is any because at the time we were not using it and did not have any use for it, though I am not sure, @wtrocki ?

That raises a good point though. Maybe the interface should have `any`, and the KnexDBDataProvider should have the typed interface. 

Not every data provider will need GraphbackContext (a custom one for example), but the interface makes it required. So I think it should be the other way around?